frame/contracts/src/lib.rs

```diff
@@ -955,9 +957,70 @@ pub mod pallet {
 		StorageValue<_, BoundedVec<DeletedContract, T::DeletionQueueDepth>, ValueQuery>;
 }
 
+/// The type of origins supported by the contracts pallet.
+#[derive(Clone)]
+pub enum ContractOrigin<T: Config> {
+	/// The origin is a regular signed account.
+	Signed(T::AccountId),
+	/// The origin is root.
+	Root,
+}
+
+impl<T: Config> ContractOrigin<T> {
+	/// Creates a new Signed ContractOrigin from an AccountId.
+	pub fn from_account_id(account_id: T::AccountId) -> Self {
+		ContractOrigin::Signed(account_id)
+	}
```

Comment:
why adding this while you can just call the inner directly?

Reply:
reason behind is that in the original [issue](https://github.com/paritytech/substrate/issues/13644) we leave the door open to having more origins in the future, some of them could potentially have account ids associated with them.
Having a function wrapping that up makes it more maintainable and open to future changes, and if those changes don't come it doesn't hurt to have it, maybe it's a little less readable but I don't think it's a big issue